OpenEdXMobile/src/main/java/org/edx/mobile/player/AudioMediaService.java

```diff
@@ -0,0 +1,526 @@
+    NotificationManager notificationManager;
+    RemoteViews notificationRemoteView;
+    IBinder iBinder = new MusicBinder();
+    boolean isRunningForeground = false;
+    boolean isBound = false;
+    boolean isNotificationShowing = false;
+    public static final String MUSIC_CHANNEL = "MusicChannel";
+    String notificationTitle, notificationMessage;
+
+    protected EnrolledCoursesResponse courseData;
+    protected String courseComponentId;
+
+
+    public static final String ACTION_MAIN_ACTIVITY = "MAIN_ACTIVITY";
+    public static final String START_FOREGROUND_SERVICE = "START_FOREGROUND_SERVICE";
+    public static final String START_SERVICE = "START_SERVICE";
+    public static final String STOP_SERVICE = "STOP_SERVICE";
+    public static final String STOP_FOREGROUND_SERVICE = "STOP_FOREGROUND_SERVICE";
+
+    public static final String PLAY_MEDIA = "PLAY_MEDIA";
+    public static final String PAUSE_MEDIA = "PAUSE_MEDIA";
+    public static final String DELETE_INTENT = "DELETE_INTENT";
+    public static final String CANCEL_INTENT = "CANCEL";
+    public static int NOTIFICATION_ID = 101;
+
+    IPlayerListener iPlayerListenerActivityCallbacks;
+    IPlayer currentPlayer;
+
+    HashMap<String , Player> connectedPlayers= new HashMap<>();
+    @Override
+    public void onCreate() {
+
+        super.onCreate();
+        notificationManager = (NotificationManager) getSystemService(NOTIFICATION_SERVICE);
+//        android.os.Debug.waitForDebugger();
+    }
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        Log.d(TAG , "START COMMAND");
+        if (intent != null)
+        {
+            String action = intent.getAction() == null ? "" : intent.getAction();
+            //Stop the notification if already shown
+            if(action.equals(START_SERVICE)) {
+                stopForegroundService(true);
+            }
+
+            //Play button clicked in notification
+            else if (action.equals(PLAY_MEDIA)) {
+                handleResumeAction(false);
+                Log.d(TAG , "RESUME/PLAY COMMAND");
+            }
+
+            //Pause button clicked in notification
+            else if (action.equals(PAUSE_MEDIA)) {
+                handlePauseAction(false);
+                Log.d(TAG , "PAUSE COMMAND");
+
+            }
+
+            //Cancel button clicked in notification
+            else if(action.equals(CANCEL_INTENT))
+            {
+                Log.d(TAG , "CANCEL COMMAND");
+                resetAllPlayers();
+                releaseAllPlayers();
+                currentPlayer = null;
+                connectedPlayers.clear();
+                stopForegroundService(true);
+                stopSelf();
+            }
+            else if(action.equals(DELETE_INTENT))
+            {
+                Log.d(TAG , "DELETE COMMAND");
+
+                if(!isBound)
```

Comment:
Extract this if-else to a separate method to avoid two-level nesting in a method